03/Product.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+export default class Product extends React.Component{
+   addToBasket = () => {
+        this.props.addToCart(this.props.product.name);
+   }
+
+   removeFromBasket = () => {
+        this.props.removeFromCart(this.props.product);
+   }
+
+    render() {
+        if(this.props.isCategory === true) {
+            if(this.props.cart.includes(this.props.product.name)) {
+                return <li>{this.props.product.name} <button disabled onClick={this.addToBasket}>BUY</button> </li>
+            }
+
+            return <li>{this.props.product.name} <button onClick={this.addToBasket}>BUY</button> </li>
+        }
+
+        if(this.props.isCart === true) {
```

Comment:
Może `else if` - skoro jest to kategorią to nie będzie koszykiem.
Nie ma potrzeby sprawdzania obu warunków :)